CHANGELOG.md

```diff
@@ -28,6 +28,26 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   **Deadline**: `jj checkout` will be deleted and become a **hard error in
   version 0.18.0**.
 
+* `jj merge` is now deprecated; use `jj new` instead.
+
+  **Rationale**: `jj merge` is functionally equivalent to `jj new`, except it
+  requires *at least* two arguments. `jj new` can create a new commit with any
+  number of parents; whether that is 1 parent &mdash; a single commit &mdash; or
+  2, or 3 parents, which would represent various merges. `jj merge` is a
+  specialization of existing functionality; it only creates "merge commits", but
+  only because "merge commit" is defined as "has at least two parents."
+
+  This makes `jj merge` functionally redundant, and also obscures the relative
+  generality of the `jj new` command, which can create a commit from any number
+  of parents. We encourage all users to adopt `jj new` for performing merges
+  instead; it is shorter, more general, and easier to remember.
+
+  `jj merge` will no longer be shown as part of `jj help`, but will still
+  function for now, emitting a warning about its deprecation.
+
+  **Deadline**: `jj merge` will be deleted and become a **hard error in
+  version 0.18.0**.
```

Comment:
It seems simpler to merge this with the item about `jj checkout` above, if you think we can make it readable enough (seems like it to me). Then we don't waste the reader's time as much by making them read it twice (or, perhaps more likely, making them compare the two sections to see if there's any surprising difference between them).